src/lib/token_tree.js

```diff
@@ -0,0 +1,92 @@
+export default class TokenTree {
+  constructor() {
+    this.rootNode = { root: true, children: [] };
```

Comment:
I can't see how the `root` flag is used. Why do you need it?

Reply:
It seemed a helpful concept, does it bother you?

Reply:
But you really ain't gonna need it (:
With this flag in place, there's no certainty about what is the root: a node that `rootNode` points to, or a node with `root` flag?
Moreover, you have to delete that root flag from sublanguage nodes.
Less code, less problems...

Reply:
Removed.